### DISCUSSION TITLE
Bulk edit aliases: delete, change mailbox, etc

### DISCUSSION THREAD

**Discussion by nguyenkims** [simple-login/app#407] [Feature Requests]
User can select multiple aliases and apply operations like delete, change mailbox, disable, etc

Moved from old roadmap https://trello.com/c/outf99Yk/63-bulk-edit-aliases-delete-change-mailbox-etc

**Comment by gritcalculator** [comment]
Yes, this would be great. Due to a spammer, I received last night 700+ e-mails to my catch-all. The system created 700+ aliases, and now I need to swipe, click remove, click yes 700+ times 🙈..

**Reply by abdullahak** [reply]
that happened with my catch-all too so now I had to disable the catch-all option there

**Comment by cphaynes** [comment]
Came here to vote on this. 

I own a 4 character .com, that is only a TLD away from a large non-profit's domain. The catch-all system, while great, is sending massive amounts of spam my way. Would love to see bulk alias management capabilities.

**Comment by pluja** [comment]
Is there any plans to add this? I need to change the mailbox from all my aliases (90+) and I will need to do it one by one

**Reply by nguyenkims** [reply]
It's indeed in the todo list. If you want to change where the aliases are forwarding, a workaround is to change the mailbox address instead of adding a new one.

**Reply by pluja** [reply]
Thank you!

**Reply by ghost** [reply]
Is there a possibility of changing the mailbox for multiple alias at once?

**Reply by RoadDragon** [reply]
> Is there any plans to add this? I need to change the mailbox from all my aliases (90+) and I will need to do it one by one

Me too. I will be doing this manually today. A bulk selection would be helpful.

**Comment by snipecelly** [comment]
Would like to see this feature for the android app as well. Thank you.

**Comment by mdbdb** [comment]
Any updates here?

This feature would be really cool.

**Comment by xtcmonster** [comment]
Seriously in need of this

**Comment by mrdigital027** [comment]
Any update on this feature request?  Hopefully soon now.  Changing hundreds of aliases isn't a fun option. :)

**Comment by gausson** [comment]
How about making a list view for alias listing in the dashboard? The card is consuming too much space in the dashboard even on a desktop and showing too much not-very-useful information. If put them into a list view with a checkbox selector, it can highly improve the effeciency of managing a long list of aliases. Beside the filter of the dashboard should be able to be filtered by domain rather than mailbox, but I think it may go into another feature request.

**Reply by DerekSeaman** [reply]
I 100% agree. I used Burnermail.io, and their dashboard is a list view. They had several sorting options as well, which made the list easy to consume and view the information I wanted. The current "card" view is very space inefficient and hard to find details that I want.

**Comment by linedpaper** [comment]
Bulk update and list view please!

**Comment by thedevelopercat** [comment]
I found a work-around:

1. Go to mailboxes and click `Delete` on the mailbox containing the aliases you want to move.
2. A pop-up will appeal containing a drop-down having `Delete aliases`.
3. Click on it and select the new inbox.

This worked for me, I hope it helps.

**Reply by github-mmfna** [reply]
Thank you for this!!

**Comment by zachvpeirce** [comment]
I recently wanted to update all of my mailboxes so I had better control over filters.

I had about 200 aliases to update, I was focused on updating just the mailboxes. I was able to update everything using the API. I used the following: 
1) https://github.com/simple-login/app/blob/master/docs/api.md#get-apiv2aliases - This got me all of the existing aliases. 
2) https://github.com/simple-login/app/blob/master/docs/api.md#patch-apialiasesalias_id - This was the update function, I used PUT instead of patch because I saw this is what the website browser does and it worked. 

In my case I needed to get the mailbox_id from the website in the URL of the mailbox setting page. 

Here was my PUT API Call: 
```
PUT /api/aliases/XXXXXXX HTTP/1.1
Host: app.simplelogin.io
Authentication: [removed]
Content-Type: application/json

{"mailbox_ids":[YYYYYYY]}
```
XXXXXXX = alias_id
YYYYYYY = mailbox_id


Generally, it was pretty easy. If you are a bit technically inclined, this can easily be done in less than an hour using a spreadsheet and postman (use postman runner to cycle through each alias).

**Comment by whatinthenameofgod** [comment]
I keep hoping this gets done. Updating hundreds of aliases is such a pain!

**Comment by hudp16** [comment]
Bulk alias select functionality, like a normal email inbox, would be a god send. Getting a 429 error because I am making 3 clicks too fast while deleting aliases is ridiculous!

**Comment by schwark** [comment]
https://github.com/schwark/simplelogin-utils

**Comment by justadityaraj** [comment]
+1. A must have feature!

**Comment by hiccups1980** [comment]
this is wish number 1 (most upvotes) in the forum. any news on this? @nguyenkims

**Comment by xgitqa** [comment]
It's been years. Is there any update on this feature request?
